### ISSUE TITLE
" java.lang.IllegalStateException: Response is committed "

### ISSUE THREAD

**Issue by Matyeu** [perwendel/spark#910]
Hey guy I have an IllegalStateException when I redirect with Filter, for example:

```java
public static Filter beforeLoad = (Request request, Response response) -> {
        if (request.url().equals("http://localhost/")) {
            System.out.println("REDIRECT");
            response.redirect("https://google.fr/");
            System.out.println("STOP");
        }
    };

before("*", beforeLoad);

get("/", (request, response) -> {
    Map<String, Object> model = new HashMap<>();
    if (request.session().attribute(Configuration.USER_SESSION) != null) {
        User user = request.session().attribute(Configuration.USER_SESSION);
        model.put("user", user);
    }
    return Render.render(request, model, "views/test.vm");
});
```

Print trace: http://i.imgur.com/QJEeMW6.png

The before() method can't cancel the response again.
I think that it would be necessary to verify if the answer is commited before execute get() method ?

**Comment by Matyeu** [comment]
For fix this I have placed halt method like that:

public static Filter beforeLoad = (Request request, Response response) -> {
        if (request.url().equals("http://localhost/")) {
            System.out.println("REDIRECT");
            response.redirect("https://google.fr/");
            halt();
            System.out.println("STOP");
        }
};

And that stoped the future request.

**Comment by jakaarl** [comment]
I think the underlying problem is possibly fixed by PR #860 ?
